### PR TITLE
Login to the application

### PR DESCRIPTION
**Pull request by mashadolya**

### CODE REVIEW

File: src/app/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}: {
+    children: React.ReactNode;
+    visible: boolean;
+    setVisible: Dispatch<SetStateAction<boolean>>;
+}) => {
+    const handleClose = () => setVisible(false);
+
+    return (
+        <S.ModalOverlay className="modal" visible={visible} onClick={handleClose}>
+            <S.ModalWrapper onClick={e => e.stopPropagation()}>
```
Comment (DmitryKolotilshikov) [inline src/app/components/Modal/Modal.tsx:17]: e.stopPropagation()  - it is better to put this expression in a separate function, for example stopProgationHandler, if you really need to stop the propagation in this case.

File: src/app/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import * as S from 'app/components/Modal/Modal.style';
+
+const Modal = ({
+    children,
+    visible,
+    setVisible,
+}: {
+    children: React.ReactNode;
+    visible: boolean;
+    setVisible: Dispatch<SetStateAction<boolean>>;
```
Comment (DmitryKolotilshikov) [inline src/app/components/Modal/Modal.tsx:11]: it is better to make the definition of types in a separate interface.
For example Modal: FC<ModalProps or IModal> = () =>

Comment (DmitryKolotilshikov) [inline src/app/components/Modal/Modal.tsx:11]: Where do use setVisible in this component?

File: src/app/modules/Authentication/components/SignInForm/SignInForm.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import Input from 'app/components/controls/Input/Input';
+import Types from 'app/components/controls/Input/types';
+
+const initialData: { [key: string]: string } = {
+    email: '',
+    password: '',
+};
+
+interface ISignInFormProps {
+    handleRegistrationForm: MouseEventHandler<HTMLAnchorElement>;
```
Comment (DmitryKolotilshikov) [inline src/app/modules/Authentication/components/SignInForm/SignInForm.tsx:11]: you can use SyntheticEvent<HTMLAnchorElement> it's more generic type

File: src/app/modules/Authentication/components/SignInForm/SignInForm.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+const SignInForm = ({ handleRegistrationForm }: ISignInFormProps) => {
+    const [form, setForm] = useState(initialData);
+
+    const handleSubmit = (e: React.SyntheticEvent) => {
```
Comment (DmitryKolotilshikov) [inline src/app/modules/Authentication/components/SignInForm/SignInForm.tsx:17]: return void

File: src/app/modules/Authentication/components/SignInForm/SignInForm.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    email: '',
+    password: '',
+};
+
+interface ISignInFormProps {
+    handleRegistrationForm: MouseEventHandler<HTMLAnchorElement>;
+}
+
+const SignInForm = ({ handleRegistrationForm }: ISignInFormProps) => {
+    const [form, setForm] = useState(initialData);
```
Comment (DmitryKolotilshikov) [inline src/app/modules/Authentication/components/SignInForm/SignInForm.tsx:15]: useState<????>(initialData)

File: src/app/modules/Authentication/components/SignInForm/SignInForm.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    const handleSubmit = (e: React.SyntheticEvent) => {
+        e.preventDefault();
+        //  onSubmit(loginData);
+    };
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (DmitryKolotilshikov) [inline src/app/modules/Authentication/components/SignInForm/SignInForm.tsx:22]: function returns void

File: src/app/modules/Authentication/components/controls/InputEmail.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React, { ChangeEventHandler } from 'react';
+import Input from 'app/components/controls/Input/Input';
+import Types from 'app/components/controls/Input/types';
+
+interface IInputProps {
+    value: string;
+    onChange: ChangeEventHandler<HTMLInputElement>;
+}
+
+const InputEmail = ({ value, onChange }: IInputProps) => {
```
Comment (DmitryKolotilshikov) [inline src/app/modules/Authentication/components/controls/InputEmail.tsx:10]: it is better to define props in a separate interface

File: src/custom.d.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+declare module '*.svg' {
+    import React = require('react');
+    export const ReactComponent: React.FC<React.SVGProps<SVGSVGElement>>;
+    const src: string;
+    export default src;
+}
```
Comment (DmitryKolotilshikov) [inline src/custom.d.ts:6]: why do we need this file?)

**Review by DmitryKolotilshikov — COMMENTED** [review]
NO TOXIC

File: src/app/components/Header/components/UserMenu/UserMenu.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react';
+import * as S from 'app/components/Header/components/UserMenu/UserMenu.style';
+import LinkButton from 'app/components/controls/LinkButton/LinkButton';
+import MENU_LINKS from 'app/components/Header/components/UserMenu/constants/menuLinks';
+import Modal from 'app/components/Modal/Modal';
+import SignInForm from 'app/modules/Authentication/components/SignInForm/SignInForm';
+
+const UserMenu = () => {
+    const [modal, setModal] = useState(true);
+    const [registrationForm, setRegistrationForm] = useState(false);
```
[the remark this answers is not in the document]
Comment (mashadolya) [reply src/app/components/Header/components/UserMenu/UserMenu.tsx]: TS automatically determines the return type in this case

File: src/app/modules/Authentication/components/SignInForm/SignInForm.tsx (hunk 4 above)
Comment (mashadolya) [reply src/app/modules/Authentication/components/SignInForm/SignInForm.tsx]: The same as above

**Comment by dpryshchep** [comment]
Please merge this one

**Pull request closed without merging**